### PR TITLE
Warn on access token reveal

### PR DESCRIPTION
**Pull request by aaronraimist**
Fixes vector-im/element-web#15367

This is the same code from @resynth1943 in #5355, just with the merge conflicts fixed and an updated screenshot

### CODE REVIEW

File: src/components/views/dialogs/AccessTokenDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2017 Vector Creations Ltd
```
Comment (jryans) [inline src/components/views/dialogs/AccessTokenDialog.tsx:2]: Hmm, remove Vector here...? Seems confusing / unneeded.

**Review by jryans — CHANGES_REQUESTED** [review]
A tiny thing to change code wise, otherwise looks good. Will need a Design review as well.

**Comment by aaronraimist** [comment]
Hmm apparently the re-request review button removes other existing requests? you'll have to add the design request back

**Comment by SimonBrandner** [comment]
@aaronraimist, yep, it seems to be a common GitHub bug....

**Review by jryans — APPROVED** [review]

**Comment by niquewoodhouse** [comment]
Thanks @aaronraimist I notice the previous version already had design feedback, the only comment being to change the body copy.  And @nadonomy has way more familiarity with these areas than I do.

So I'm tempted to just approve it because it's a good iteration but I have a couple of additional notes.  I welcome your opinions on these notes, I'm not too familiar with the users who are likely to interact with this, or why someone would, so please share your thoughts.  I also don't know the effort required to do them.

**Notes**
- I wondered if keeping the `<click to reveal>` made sense, and changing it to `<click to hide>` to enable hiding after it's been shown.  It's just we're now saying this is a really important thing, don't share it, and then there's no visual way to hide it.   I think _some_ people might get a bit concerned by that, and would be re-assured they can hide it again.  If that's easy, it seems a simple iteration to make this a bit more friendly if non-technical people ever need to do this.

- My final note was about the button, I don't quite get why the button is red, it's not an error?  I can see for security flows there's a few times there's red buttons used for primary actions so maybe it's the right thing to do for security flows (@jryans do you know if that's true) but my personal bias says it makes it hard to know what to do.  My second point on the button is "Ok" might not be the right copy for the primary action, "reveal access token" or "reveal" or "reveal token"  or "show access token".  They just feel a bit more accurate as to what's happening next, whereas "Ok" just sounds like "Understood".


Thanks

**Comment by aaronraimist** [comment]
@niquewoodhouse right now the access token is hidden again when you navigate away from the Help & About section of settings. When you come back it will be hidden.

How about we just simplify this a bit and remove the dialog box and the custom click to hide/reveal button and make the access token hidden behind a standard [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) element. Something like this:

<details>
    <summary>Access Token</summary>
    <b>Your access token gives full access to your account. Do not share it with anyone.</b>  

   MDA123ABC...
</details>

**Comment by niquewoodhouse** [comment]
@aaronraimist I really really like that!  You wierdly read my mind - I wanted to propose this as the change originally, but felt there had already been lots of good done so didn't want to suggest undoing a lot of it.  

I noticed there's already some UI for expanding/collapsing in settings, under appearance.

image

And there's these in security and privacy.

image

I prefer the ones in security and privacy, they look more like the proposal and reflect the room list sections.

Thanks for coming back with this.

**Comment by aaronraimist** [comment]
Okay I think this is ready to go. Here is what it looks like now:

Collapsed:


Expanded:

**Comment by aaronraimist** [comment]
(someone will need to re-request design review)

**Review by jryans — APPROVED** [review]
Looks good to me code wise, thanks! 😄

**Comment by niquewoodhouse** [comment]
That's so much better, thank you.

- Couldn't tell if this is a netlify thing, or my browser but, after I copy the field, the copied! text remains when I leave the settings and go to other areas in Element.  Maybe it's just a thing with the copy field component but just raising in case its not.

image

- Quick win: It would be nice if we could bold the following sentences.  `Your access token gives full access to your account. Do not share it with anyone.`  Otherwise this warning is quite muted and can go unseen.  There is some precedence in settings for bolding key information, such as your identity server.

Thank you again

**Comment by aaronraimist** [comment]
@niquewoodhouse Before it was only closing the tooltip if you moved your mouse. It should now close the tooltip if you close the settings screen in some other way such as pressing escape. Let me know if it closes for you now.

**Review by niquewoodhouse — APPROVED** [review]
Thanks for this, works great and is much better.

**Review by jryans — APPROVED** [review]
Looks good, thanks for working on this! 😄

**Pull request merged**